Modified content in Level 2 for video script (#218)
pupilfirst/wd301-scripts: typescript-fundamentals/executing-ts-code/README.md

> **[INTRODUCTION]**

Reviewer: ```suggestion
## INTRODUCTION
```

Hello! In this level, we'll tackle a fundamental aspect of TypeScript: how to execute and run TypeScript code.

If you're new to TypeScript, you might be wondering how to take the code you've written and actually run it in a web browser or another environment. Let's go over the steps you need to follow to do just that.

> **[TYPESCRIPT COMPILATION]**

Reviewer: ```suggestion
## TypeScript Compilation
```

@developedbysaad Generally, format headings as headings (##, ### and so on) instead of bold and no all-caps.

> Action: Display the text "TypeScript Compilation" on the screen.

First, it's important to understand that TypeScript code cannot be directly run in a web browser or other environments. TypeScript is a compiled language, which means it needs to be transformed into plain JavaScript before it can be executed. But don't worry, TypeScript makes this process easy.

> **[STEP 1: Install TypeScript Compiler]**

Reviewer: ```suggestion
### Step 1: Install TypeScript Compiler
```

> Action: Display the command to install the TypeScript compiler.

> The first step is to install the TypeScript compiler on your machine. You can do this by running the following command:

Reviewer: @developedbysaad Add where (path) this command is supposed to be run.

```shell
npm install -g typescript
```

This installs TypeScript globally on your system.

> **[STEP 2: Create a TypeScript File]**

Reviewer: ```suggestion
### Step 2: Create a TypeScript file
```

> Action: Show the creation of an `index.ts` file and add code.

> Next, create a new file in your working directory and name it `index.ts`. Inside this file, you can write your TypeScript code. For example:

Reviewer: ```suggestion
Next, create a new file in your working directory, i.e. `smarter-tasks` folder and name it `index.ts`. Inside this file, you can write your TypeScript code. For example:
```

```typescript
console.log("This is a TypeScript file!!!");
```

> **[STEP 3: Compile TypeScript Code]**

Reviewer: ```suggestion
### Step 3: Compile TypeScript code
```

> Action: Demonstrate how to compile TypeScript code.

> Once you have the TypeScript compiler installed, you can use it to compile your TypeScript code into JavaScript. To do this, navigate to the directory containing your TypeScript files and run the following command:

Reviewer: ```suggestion
Once you have the TypeScript compiler installed, you can use it to compile your TypeScript code into JavaScript. To do this, navigate to `smarter-tasks` directory that contains the file you just created and run the following command:
```

```shell
tsc index.ts
```

This command will compile your `index.ts` file into a corresponding `index.js` file. You can also compile multiple TypeScript files at once by specifying them as additional arguments.

> **[STEP 4: Run JavaScript Code]**

Reviewer: ```suggestion
### Step 4: Run JavaScript code
```

> Action: Show how to run the generated JavaScript code in a web browser.

Now that your TypeScript code is compiled into JavaScript, you can run it just like any other JavaScript code. If you're running it in a web browser, include the generated `.js` file in your HTML file like this:

```html
<script src="index.js"></script>
```

Alternatively, if you want to run the generated JavaScript code using Node.js, you can use the following command:

```shell
node index.js
```

> **[ALTERNATIVE: Using ts-node]**

Reviewer: ```suggestion
## Alternative approach Using ts-node
```

> Action: Explain an alternative method using `ts-node`.

While the previous steps work, in most scenarios, we use Node.js bundled with other packages to run our TypeScript applications.

Alternatively, you can use an external npm package called `ts-node` to compile and run your TypeScript code in a single command.

- Install and use the `ts-node` package by running:

```shell
npm install -g ts-node
```

- Once installed, you can directly execute your TypeScript code by running:

```shell
ts-node index.ts
```

This simplifies the process of running TypeScript code.

> **[CONCLUSION]**

Reviewer: ```suggestion
## Conclusion
```
